src/libs/ajax/User.ts

```diff
@@ -0,0 +1,398 @@
+      try {
+        const response = await fetchSam(
+          'register/user/v1/termsofservice',
+          _.mergeAll([authOpts(), { signal, method: 'DELETE' }])
+        );
+        return response.json();
+      } catch (error: any) {
+        if (error.status !== 404) {
+          throw error;
+        }
+      }
+    },
+
+    getTermsOfServiceComplianceStatus: async (): Promise<SamUserTosComplianceStatusResponse | null> => {
+      try {
+        const res = await fetchSam(
+          'register/user/v2/self/termsOfServiceComplianceStatus',
+          _.merge(authOpts(), { signal })
+        );
+        return res.json();
+      } catch (error: any) {
+        if (error.status === 404 || error.status === 403) {
+          return null;
+        }
+        throw error;
+      }
+    },
+
+    getPrivacyPolicy: async (): Promise<string> => {
+      const response = await fetchSam('privacy/text', _.merge(authOpts(), { signal }));
+      return response.text();
+    },
+
+    // If you are making changes to the Support Request Modal, make sure you test the following:
+    // 1. Submit a ticket via Terra while signed in and signed out
+    // 2. Check the tickets are generated on Zendesk
+    // 3. Reply internally (as a Light Agent) and make sure an email is not sent
+    // 4. Reply externally (ask one of the Comms team with Full Agent access) and make sure you receive an email
+    createSupportRequest: ({
+      name,
+      email,
+      currUrl,
+      subject,
+      type,
+      description,
+      attachmentToken,
+      emailAgreed,
+      clinicalUser,
+    }) => {
+      return fetchOk(
+        'https://support.terra.bio/api/v2/requests.json',
+        _.merge(
+          { signal, method: 'POST' },
+          jsonBody({
+            request: {
+              requester: { name, email },
+              subject,
+              // BEWARE changing the following ids or values! If you change them then you must thoroughly test.
+              custom_fields: [
+                { id: 360012744452, value: type },
+                { id: 360007369412, value: description },
+                { id: 360012744292, value: name },
+                { id: 360012782111, value: email },
+                { id: 360018545031, value: emailAgreed },
+                { id: 360027463271, value: clinicalUser },
+              ],
+              comment: {
+                body: `${description}\n\n------------------\nSubmitted from: ${currUrl}`,
+                uploads: [`${attachmentToken}`],
+              },
+            },
+          })
+        )
+      );
+    },
+
+    uploadAttachment: async (file) => {
+      const res = await fetchOk(`https://support.terra.bio/api/v2/uploads?filename=${file.name}`, {
+        method: 'POST',
+        body: file,
+        headers: {
+          'Content-Type': 'application/binary',
+        },
+      });
+      return (await res.json()).upload;
+    },
+
+    firstTimestamp: (): RexFirstTimestampResponse => {
```

Comment:
This actually returns a Promise since it fetches from Rex.

```suggestion
    firstTimestamp: (): Promise<RexFirstTimestampResponse> => {
```